### PR TITLE
Removing `more_itertools` dependency.

### LINKED ISSUE #20508: more_itertools required for Whisper normaliser
### System Info

- `transformers` version: 4.25.0.dev0
- Platform: macOS-12.5-arm64-arm-64bit
- Python version: 3.8.9
- Huggingface_hub version: 0.10.0
- PyTorch version (GPU?): 1.11.0 (False)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): 0.5.1 (cpu)
- Jax version: 0.3.6
- JaxLib version: 0.3.5
- Using GPU in script?: no
- Using distributed or parallel set-up in script?: no

### Who can help?

cc @ArthurZucker

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

The English normaliser for Whisper relies on the `more_itertools` package.

It is imported here:
https://github.com/huggingface/transformers/blob/761b3fad922310457003af2fea6c447768676c8d/src/transformers/models/whisper/english_normalizer.py#L23-L24

And used here:
https://github.com/huggingface/transformers/blob/761b3fad922310457003af2fea6c447768676c8d/src/transformers/models/whisper/english_normalizer.py#L243

Since we import `more_itertools` under the if statement `if is_more_itertools_available()`, the normaliser **fails** if `more_itertools` is **not** installed.

```python
from transformers import WhisperTokenizer

tokenizer = WhisperTokenizer.from_pretrained("openai/whisper-tiny.en")

tokenizer._normalize("the cat")
```

```
File "/home/user/.local/lib/python3.8/site-packages/transformers/models/whisper/tokenization_whisper.py", line 485, in _normalize
    return normalizer(text)
  File "/home/user/.local/lib/python3.8/site-packages/transformers/models/whisper/english_normalizer.py", line 593, in __call__
    s = self.standardize_numbers(s)
  File "/home/user/.local/lib/python3.8/site-packages/transformers/models/whisper/english_normalizer.py", line 497, in __call__
    s = " ".join(word for word in self.process_words(s.split()) if word is not None)
  File "/home/user/.local/lib/python3.8/site-packages/transformers/models/whisper/english_normalizer.py", line 497, in 
    s = " ".join(word for word in self.process_words(s.split()) if word is not None)
  File "/home/user/.local/lib/python3.8/site-packages/transformers/models/whisper/english_normalizer.py", line 243, in process_words
    for prev, current, next in windowed([None] + words + [None], 3):
NameError: name 'windowed' is not defined
```

IMO this is a pretty cryptic error message for the user. Perhaps we can add a warning that `more_itertools` is required for the normaliser? Even better, we implement the `windowed` function ourselves to prevent an extra library dependency that we use for one function.

### Expected behavior

Good: warning that `more_itertools` is not installed
Better: implement the `windowed` function ourselves

### PR DESCRIPTION
**Pull request by Narsil**
# What does this PR do?

Removes `more_itertools` optional dependency.
Fixes  #20508

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/transformers/models/whisper/english_normalizer.py [hunk 1]
```diff
@@ -14,17 +14,10 @@
 # limitations under the License.
 
 import re
+import unicodedata
```
Comment (ArthurZucker) [inline src/transformers/models/whisper/english_normalizer.py:17]: I am guessing we already have this as a dependency?

Comment (Narsil) [reply src/transformers/models/whisper/english_normalizer.py:17]: It got moved by `isort` that's it.

**Comment by Narsil** [comment]
I don't think the failure is linked in any way to this PR, is it ?

File: examples/research_projects/vqgan-clip/requirements.txt [hunk 2]
```diff
@@ -4,7 +4,6 @@ icecream
 imageio
 lpips
 matplotlib
-more_itertools
```
Comment (sgugger) [inline examples/research_projects/vqgan-clip/requirements.txt:7]: Research projects can have other dependencies, I'd leave it there.

**Review by sgugger — APPROVED** [review]
Thanks for the cleanup!

**Pull request merged**